samples/widgets/headerctrl.cpp

```diff
@@ -200,19 +200,20 @@ void HeaderCtrlWidgetsPage::RecreateWidget()
     m_header = new wxHeaderCtrlSimple(this, wxID_ANY,
                                       wxDefaultPosition, wxDefaultSize,
                                       flags);
-    wxASSERT(WXSIZEOF(m_colSettings) == 2);
-    wxHeaderColumnSimple col1("First", 100, GetColumnAlignmentFlag(0), GetColumnStyleFlags(0));
-    if ( m_colSettings[0].chkWithBitmap->IsChecked() )
-    {
-        col1.SetBitmap(wxArtProvider::GetIcon(wxART_ERROR, wxART_BUTTON));
-    }
-    m_header->AppendColumn(col1);
-    wxHeaderColumnSimple col2("Second", 200, GetColumnAlignmentFlag(1), GetColumnStyleFlags(1));
-    if ( m_colSettings[1].chkWithBitmap->IsChecked() )
+
+    for ( int i = 0; i < (int)WXSIZEOF(m_colSettings); i++ )
     {
-        col2.SetBitmap(wxArtProvider::GetIcon(wxART_QUESTION, wxART_BUTTON));
+        wxHeaderColumnSimple col(wxString::Format("Column %d", i + 1),
+                                 FromDIP(100),
+                                 GetColumnAlignmentFlag(i),
+                                 GetColumnStyleFlags(i));
+        if ( m_colSettings[i].chkWithBitmap->IsChecked() )
+        {
+            wxArtID artID = i % 4 == 0 ? wxART_ERROR : (i % 4 == 1 ? wxART_QUESTION : (i % 4 == 2 ? wxART_WARNING : wxART_INFORMATION));
```

Comment:
```suggestion
            wxArtID artID;
            switch ( i % 4 )
            {
                case 0: artID = wxART_ERROR; break;
                case 1: artID = wxART_QUESTION; break;
                case 2: artID = wxART_WARNING; break;
                case 3: artID = wxART_INFORMATION; break;
            }
```
or maybe
```
const wxArtID icons[] = { wxART_ERROR, wxART_QUESTION, wxART_WARNING, wxART_INFORMATION };
wxArtID artID = icons[i % WXSIZEOF(icons)];
```

Both of those seem more readable to me.

Reply:
I, being in the "never use literals in the code" camp, would also suggest replacing the literal `4` everywhere, including  `m_colSettings` declaration, with a constant or sizeof expression.

Reply:
Removed literals, only one define remaining for the number of columns. This makes it easy to quickly adjust the sample to show more or less columns.